### ISSUE TITLE
[BUG] Documentation about rules is leading to error    Definition for rule 'vitest/valid-title' was not found

### ISSUE THREAD

**Issue by SalahAdDin** [vitest-dev/eslint-plugin-vitest#510]
We are updating our configuration to the new plugin with the latest version.

We followed the [documentation](https://github.com/vitest-dev/eslint-plugin-vitest?tab=readme-ov-file#usage) for legacy code and we defined the configuration as follows:
```js
    {
      "files": ["src/**/*.test.[tj]s?(x)"],
      "excludedFiles": ["src/**/*.e2e.test.[tj]s?(x)"],
      "extends": [
        "plugin:jest-dom/recommended",
        "plugin:testing-library/react",
        "plugin:@vitest/legacy-all"
      ],
      "rules": {
        "@vitest/valid-title": [
          "error",
          {
            "mustMatch": {
              "it": [
                "^should.*when.+$",
                "Test title must include 'should' and 'when'"
              ]
            }
          }
        ]
      }
    },
```

But it leads to a bug since we are using `@vitest` plugin.

It can be fixed by using `@vitest` on the rules. Hence, the documentation for legacy code must be updated.

There is no problem with the new ESLint configuration.

**Comment by veritem** [comment]
I don't know if it makes sense to update docs since the legacy will eventually deprecated.

**Comment by SalahAdDin** [comment]
> I don't know if it makes sense to update docs since the legacy will eventually deprecated.

But It is not deprecated yet.
